src/bitpay_sdk/utils/rest_cli.py

```diff
@@ -0,0 +1,171 @@
+"""
+HTTP methods
+"""
+import json
+import urllib
+import requests
+
+from .. import env
+from ..exceptions.bitpay_exception import BitPayException
+from .key_utils import get_compressed_public_key_from_pem, sign
+
+
+class RESTcli:
+    __client = ""
+    __baseurl = ""
+    __eckey = ""
+    __identity = ""
+    __proxy = ""
+
+    def __init__(self, environment, eckey, proxy=None):
+        self.__eckey = eckey
+        self.__baseurl = env.TESTURL if environment == env.TEST else env.PRODURL
+        self.__proxy = proxy
+        self.init()
+
+    def init(self):
+        try:
+            config = {
+                'base_url': self.__baseurl,
+                'defaults': {
+                    'headers': {
+                        'x-accept-version': env.BITPAYAPIVERSION,
+                        'x-bitpay-plugin-info': env.BITPAYPLUGININFO,
+                        'x-bitpay-api-frame': env.BITPAYAPIFRAME,
+                        'x-bitpay-api-frame-version': env.BITPAYAPIFRAMEVERSION,
+                    },
+                }
+            }
+            if self.__proxy != '':
+                config['proxy'] = self.__proxy
+
+        except BitPayException as e:
+            print(e)
+
+    def post(self, uri, form_data, signature_required=True):
+        """
+        :param uri: Url at which user wants to post the data
+        :param form_data: the data to be posted
+        :param signature_required: Signature of the full request URL concatenated
+        with the request body
+        :return: json response
+        """
+        full_url = self.__baseurl + uri
+        form_data = json.dumps(form_data)
+
+        xidentity = get_compressed_public_key_from_pem(self.__eckey)
```

Comment:
this only needs to be run if the signature is required, please move it inside "if signature_required:" for all methods